server/controllers/profile.js

```diff
@@ -63,12 +62,63 @@ exports.retriveImgUrls = asyncHandler(async (req, res) => {
       profile.uploadedImages.push(imageUrl);
       await profile.save();
     }
-  }
+  };
   saveImagesToDb(results);
 
-  //delete images in the uploads folder 
-  await Promise.all(files.map(async (file) => {
-    await unlinkFile(file.path);
-  }));
+  //delete images in the uploads folder
+  await Promise.all(
+    files.map(async (file) => {
+      await unlinkFile(file.path);
+    })
+  );
   res.status(201).json("image uploaded");
-});
\ No newline at end of file
+});
+
+// @route POST /profile/upload-avatar
+// @desc Upload a profile photo to server
+// @access Private
+exports.retrieveAvatarUrl = asyncHandler(async (req, res) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+  if (!profile) {
+    res.status(404);
+    throw new Error("Profile doesn't exist");
+  }
+  const file = req.file;
+  const result = await uploadFileToS3(file);
+  profile.photo = result.Location;
+  await profile.save();
+  await unlinkFile(file.path);
+  res.status(201).json({
+    imageKey: result.Key
+  });
+});
+
+// @route GET /profile/photo/:key
+// @desc Get a profile photo from sever
+// @access Private
+exports.getAvatar = async (req, res) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+  if (!profile) {
+    res.status(404);
+    throw new Error("Profile doesn't exist");
+  }
+  res.status(201).json({
+    imagePath: profile.photo
+  });
+};
+
+// @route DELETE /profile/photo/:key
+// @desc Delete a profile photo from sever
+// @access Private
+exports.deleteProfilePhoto = async (req, res) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+  if (!profile) {
+    res.status(404);
+    throw new Error("Profile doesn't exist");
+  }
+  const key = req.params.key;
+  const result = await deleteFile(key);
+  profile.photo = "";
+  await profile.save();
+  res.status(201).json("profile photo deleted");
```

Comment:
Similar to my other  comment, we should say anything was created on a delete route. `200` or `204` would be more appropriate here. If you choose `204` then you shouldn't respond with anything but just the status code. `200` will do just fine 👍🏻